hubblestack/extmods/hubble_mods/osquery.py

```diff
@@ -0,0 +1,217 @@
+'''
+Flexible Data Gathering: osquery
+
+This module allows for running osquery queries
+
+Mandatory param:
+    query -  The query which needs to be run
+
+Optional param:
+    cast_to_string - Default (False) - Specifies if the command output (ex. list of dictionaries)  needs to be type cast to string
+
+    osquery_args - string of args to pass to osquery. Note that the ``--read_max`` and ``--json`` args are already included.
+
+    osquery_path - specific path to the osquery binary
+
+    format_chained - Default (True) - Whether to call ``.format(chained)`` on the query. Set to False if you want to avoid having to escape curly braces.
+
+Audit Example:
+---------------
+check_unique_id:
+  description: 'osquery check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:CentOS*Linux-7'
+      hubble_version: '>3 AND <7 AND <8'
+      module: osquery
+      items:
+        - args:
+            query: 'select count(*) as count from (select pn.pid_namespace from process_namespaces pn JOIN processes pr on pn.pid=pr.pid where pr.name LIKE "%dockerd%" INTERSECT select distinct(pn.pid_namespace) from process_namespaces pn JOIN (select dcp.pid from docker_container_processes dcp JOIN docker_containers dc ON dcp.id = dc.id) x on pn.pid=x.pid);'
+          comparator:
+            type: list
+            size: 1
```

Comment:
We should put a meaningful comparator. Since it will be list of dictionaries.
We can put a match_any or match_all here

Reply:
done